src/labkey/query/Rows.ts

```diff
@@ -309,27 +353,33 @@ export interface SaveRowsOptions extends RequestCallbackOptions<SaveRowsResponse
  */
 export function saveRows(options: SaveRowsOptions): XMLHttpRequest {
     // Nick: I've elected to comment this out as saveRows never supported the same argument
-    // pattern as other endpoints due to the different nature of it's arguments (e.g. doesn't take a
+    // pattern as other endpoints due to the different nature of its arguments (e.g. doesn't take a
     // schema/query but rather commands, etc). As a result, the object would not match what is expected.
     // if (arguments.length > 1) {
     //     options = queryArguments(arguments);
     // }
-
-    return request({
+    const jsonData = {
+        apiVersion: options.apiVersion,
+        commands: options.commands,
+        containerPath: options.containerPath,
+        extraContext: options.extraContext,
+        transacted: options.transacted,
+        validateOnly: options.validateOnly,
+    };
+    const requestOptions: RequestOptions = {
         url: buildURL('query', 'saveRows.api', options.containerPath),
         method: 'POST',
-        jsonData: {
-            apiVersion: options.apiVersion,
-            commands: options.commands,
-            containerPath: options.containerPath,
-            extraContext: options.extraContext,
-            transacted: options.transacted,
-            validateOnly: options.validateOnly,
-        },
         success: getCallbackWrapper(getOnSuccess(options), options.scope),
         failure: getCallbackWrapper(getOnFailure(options), options.scope, true),
         timeout: options.timeout,
-    });
+    };
+
+    const form = bindSaveRowsData(jsonData);
```

Comment:
We should consider opting into this behavior using a `autoFormFileData` bit in the options (like we did with insert rows). We don't want to unintentionally break some command(s) that are not strictly shaped like we are expecting. As much as TypeScript helps we aren't guaranteed anything by JS callers. 

Then, we can have our application calls to `saveRows()` specify this bit as true. There is already a wrapper in `api.ts` in `@labkey/components` and we would update all application calls to go through that.

Reply:
I do not understand why any callers would be passing a different shape of data than our apps? Especially considering everything I am doing here is type safe. What kind of situations are you trying to be defensive against here?